libdirections-hybrid/src/main/java/com/mapbox/navigation/route/hybrid/MapboxHybridRouter.kt

```diff
@@ -1,23 +1,150 @@
+                    callback?.onFailure(throwable)
+                }
+                false -> {
+                    isOnBoardRouterCalled = true
+                    options?.let { options ->
+                        onBoardRouter.getRoute(options, this)
+                    }
+                }
+            }
+        }
+
+        /**
+         * This method is equivalent to calling .getRoute() with the additional parameter capture
+         */
+        override fun execute(routeOptions: RouteOptions, clientCallback: Router.Callback) {
+            isOnBoardRouterCalled = false
+            options = routeOptions
+            callback = clientCallback
+            offBoardRouter.getRoute(routeOptions, this)
+        }
+    }
+
+    // On Board router used if an internet connection is not available.
+    private class OnBoardRouterHandler(
+        private val offBoardRouter: Router,
+        private val onBoardRouter: Router
+    ) : RouterDispatchInterface, Router.Callback {
+        private var isOffBoardRouterCalled = false
+        private var options: RouteOptions? = null
+        private var callback: Router.Callback? = null
+
+        override fun onResponse(routes: List<DirectionsRoute>) {
+            callback?.onResponse(routes)
+        }
+
+        /**
+         * onFailure is used as a fail-safe. If the initial call to onBoardRouter.getRoute()
+         * fails, it is assumed that the offBoardRouter may be available. The call is made to the offBoardRouter.
+         * The error returns remains the same as in the first call, but the flag value has changed. This time a failure
+         * is propagated to the client. In short, call the onBoardRouter. If it fails call the offBoardRouter,
+         * if that fails propagate the exception
+         */
+        override fun onFailure(throwable: Throwable) {
+            when (isOffBoardRouterCalled) {
+                true -> {
+                    isOffBoardRouterCalled = false
+                    callback?.onFailure(throwable)
+                }
+                false -> {
+                    isOffBoardRouterCalled = true
+                    options?.let { options ->
+                        offBoardRouter.getRoute(options, this)
+                    }
+                }
+            }
+        }
+        override fun execute(routeOptions: RouteOptions, clientCallback: Router.Callback) {
+            isOffBoardRouterCalled = false
+            options = routeOptions
+            callback = clientCallback
+            onBoardRouter.getRoute(routeOptions, this)
+        }
+    }
+
     override fun getRoute(
         routeOptions: RouteOptions,
         callback: Router.Callback
-    ) = Unit
+    ) {
+        routeDispatchHandler.get().execute(routeOptions, callback)
+    }
 
     override fun cancel() {
-        onboardRouter.cancel()
-        offboardRouter.cancel()
+        onBoardRouter.cancel()
+        offBoardRouter.cancel()
```

Comment:
Typo (was correct before)

```suggestion
        offboardRouter.cancel()
```